Code/GraphMol/Wrap/Sgroup.cpp

```diff
@@ -0,0 +1,124 @@
+             "returns whether or not a particular property exists")
+        .def("GetProp",
+             (std::string(RDProps::*)(const std::string &) const) &
+                 SGroup::getProp<std::string>,
+             "returns the value of a particular property")
+        .def("GetIntProp",
+             (int (RDProps::*)(const std::string &) const) &
+                 SGroup::getProp<int>,
+             "returns the value of a particular property")
+        .def("GetUnsignedProp",
+             (unsigned int (RDProps::*)(const std::string &) const) &
+                 SGroup::getProp<unsigned int>,
+             "returns the value of a particular property")
+        .def("GetDoubleProp",
+             (double (RDProps::*)(const std::string &) const) &
+                 SGroup::getProp<double>,
+             "returns the value of a particular property")
+        .def("GetBoolProp",
+             (bool (RDProps::*)(const std::string &) const) &
+                 SGroup::getProp<bool>,
+             "returns the value of a particular property")
+        .def("GetPropNames", &SGroup::getPropList,
+             (python::arg("self"), python::arg("includePrivate") = false,
+              python::arg("includeComputed") = false),
+             "Returns a list of the properties set on the SGroup.\n\n")
+        .def("GetPropsAsDict", GetPropsAsDict<SGroup>,
+             (python::arg("self"), python::arg("includePrivate") = true,
+              python::arg("includeComputed") = true),
+             "Returns a dictionary of the properties set on the SGroup.\n"
+             " n.b. some properties cannot be converted to python types.\n")
+#if 0
+        .def("SetProp",
+             MolSetProp<SGroup, std::string>,
+             "returns the value of a particular property")
+        .def("SetIntProp",
+             (int (RDProps::*)(const std::string &, int) const) &
+                 SGroup::setProp<int>,
+             "returns the value of a particular property")
+        .def("SetUnsignedProp",
+             (unsigned int (RDProps::*)(const std::string &, unsigned int) const) &
+                 SGroup::setProp<unsigned int>,
+             "returns the value of a particular property")
+        .def("SetDoubleProp",
+             (double (RDProps::*)(const std::string &, double) const) &
+                 SGroup::setProp<double>,
+             "returns the value of a particular property")
+        .def("SetBoolProp",
+             (bool (RDProps::*)(const std::string &, bool) const) &
+                 SGroup::setProp<bool>,
+             "returns the value of a particular property")
+#endif
+        ;
+    python::def("GetMolSGroups", &getMolSGroups,
+                "returns the SGroups for a molecule (if any)");
+    python::def("ClearMolSGroups", &clearMolSGroups,
+                "removes all SGroups from a molecule (if any)");
+    // FIX: needs something tying the lifetime to the mol

```

Comment:
I wonder if we should do that at the C++ level. Like have SGroups store a shared_ptr to the ROMol.

Reply:
They already have `getOwningMol()`, but that's a bare pointer.
Do you see a likely lifetime problem with SGroups and molecules?

Reply:
If the molecule is deleted and someone is holding onto the sgroups, there will likely be a seg fault.

I think this is what: return_internal_reference is for in boost::python

@d-b-w if the SGroup has a shared_ptr this only really works if shared pointers are used everywhere.  For example:

```
ROMol *m = MolFromMolBlock(....);
vector<SGroup> groups = getSGroups(m);
del m;
groups[0].getOwningMol();
```

Won't work unless m and the sgroup internals are the same shared_ptr.

Reply:
I.e. MolFromMolBlock would need to return the shared_ptr<ROMol> used in the sgroup.